vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -89,182 +36,72 @@
-    }
+    protected abstract ILinkDisplay getDisplayLink(String linkDisplayId) throws LinkDoesNotExistsException;
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * edu.asu.diging.vspace.core.services.impl.ILinkManager#getSpaceLinkDisplays(
-     * java.lang.String)
-     */
-    @Override
-    public List<ISpaceLinkDisplay> getSpaceLinkDisplays(String spaceId) {
-        return new ArrayList<>(spaceLinkDisplayRepo.findSpaceLinkDisplaysForSpace(spaceId));
-    }
+    protected abstract ILink getLink(String linkId) throws LinkDoesNotExistsException;
 
-    /*
-     * ===== External links =====
-     */
+    protected abstract ILink createLinkObject(String title, String id, IVSpaceElement target, String linkLabel);
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * edu.asu.diging.vspace.core.services.impl.ILinkManager#getExternalLinkDisplays
-     * (java.lang.String)
-     */
-    @Override
-    public List<IExternalLinkDisplay> getExternalLinkDisplays(String spaceId) {
-        return new ArrayList<>(externalLinkDisplayRepo.findExternalLinkDisplaysForSpace(spaceId));
-    }
+    protected abstract IVSpaceElement getTarget(String linkedId);
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * edu.asu.diging.vspace.core.services.impl.ILinkManager#createExternalLink(java
-     * .lang.String, edu.asu.diging.vspace.core.model.ISpace, float, float,
-     * java.lang.String)
-     */
-    @Override
-    public IExternalLinkDisplay createExternalLink(String title, ISpace source, float positionX, float positionY,
-            String externalLink, DisplayType displayType, byte[] linkImage, String imageFilename)
-            throws ImageCouldNotBeStoredException, SpaceDoesNotExistException {
-        // we need this to fully load the space
-        source = spaceManager.getSpace(source.getId());
+    public void spaceValidation(String id) throws SpaceDoesNotExistException{
+        ISpace source = spaceManager.getSpace(id);
         if (source == null) {
             throw new SpaceDoesNotExistException();
         }
-        IExternalLink link = externalLinkFactory.createExternalLink(title, source, externalLink);
-        externalLinkRepo.save((ExternalLink) link);
+    }
 
-        IExternalLinkDisplay display = externalLinkDisplayFactory.createExternalLinkDisplay(link);
-        display.setPositionX(positionX);
-        display.setPositionY(positionY);
-        display.setName(title);
-        display.setType(displayType != null ? displayType : DisplayType.ARROW);
+    public void linksValidation(Optional link) throws LinkDoesNotExistsException {
+        if(!link.isPresent()) {
+            throw new LinkDoesNotExistsException("Link Does Not Exists");
+        }
+    }
 
+    public void setDisplayProperties(ILinkDisplay linkDisplay,float positionX,float positionY,int rotation, DisplayType displayType, byte[] linkImage, String imageFilename) throws ImageCouldNotBeStoredException {
```

Comment:
is that used anywhere outside of the manager classes? if not, it should not be public.